providers/mock/mock.go

```diff
@@ -0,0 +1,206 @@
+package mock
+
+import (
+	"log"
+	"time"
+
+	"github.com/virtual-kubelet/virtual-kubelet/providers"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// MockProvider implements the virtual-kubelet provider interface and stores pods in memory.
+type MockProvider struct {
+	nodeName           string
+	operatingSystem    string
+	daemonEndpointPort int32
+	pods               []*v1.Pod
```

Comment:
shall we change the type to map[string]*v1.Pod, so that we can support DeletePod easily? The key can be "{namespace}-{podName}".

Reply:
Great idea! Ill make it happen.